### ISSUE TITLE
Implement new Concept Exercise: records

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#865]
This issue describes how to implement the `records` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Records in F#.

## Learning objectives

- Know what a record is.
- Know when records should be used.
- Know how to define records.
- Know that records have structural equality.
- Know how to deconstruct records (including as parameters).
- Know how to pattern match on records.
- Know that records are immutable.
- Know how to return a modified copy of a record.

## Out of scope

- `struct` records.
- Know how type inference works for records.
- Adding members to records.
- Mutually recursive records.
- Know the difference between records and classes.
- Memory and performance characteristics.

## Concepts

The Concepts this exercise unlocks are:

- `records`: know what a record is; know when records should be used; know how to define records; know that records have structural equality; know how to deconstruct records (including as parameters); know how to pattern match on records; know that records are immutable; know how to return a modified copy of a record.

## Prerequisites

This exercise's prerequisites Concepts are:

- `pattern-matching`: know how to do pattern matching.

Any data types used in this exercise (e.g. `strings`) should also be added as prerequisites.

### Hints

- [Records][records]: introduction to records.

### After

- [Records][records]: introduction to records.
- [Records introduction][records-introduction]: another record introduction, but digs a bit deeper.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/types/record.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[records]: https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/records
[records-introduction]: https://fsharpforfunandprofit.com/posts/records/
